src/NuGet.Clients/NuGet.PackageManagement.UI/ViewModels/PackageDetailsTabViewModel.cs

```diff
@@ -0,0 +1,113 @@
+using System;
+using System.Collections.ObjectModel;
+using System.ComponentModel;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Shell;
+using NuGet.VisualStudio;
+using NuGet.VisualStudio.Internal.Contracts;
+
+namespace NuGet.PackageManagement.UI.ViewModels
+{
+    public class PackageDetailsTabViewModel : ViewModelBase, IDisposable
+    {
+        private bool _disposed = false;
+
+        private bool _readmeTabEnabled;
+
+        public ReadmePreviewViewModel ReadmePreviewViewModel { get; private set; }
+
+        public DetailControlModel DetailControlModel { get; private set; }
+
+        public ObservableCollection<TitledPageViewModelBase> Tabs { get; private set; }
+
+        private TitledPageViewModelBase _selectedTab;
+        public TitledPageViewModelBase SelectedTab
+        {
+            get => _selectedTab;
+            set
+            {
+                SetAndRaisePropertyChanged(ref _selectedTab, value);
+            }
+        }
+
+        public PackageDetailsTabViewModel()
+        {
+            _readmeTabEnabled = true;
+            Tabs = new ObservableCollection<TitledPageViewModelBase>();
+        }
+
+        public PackageMetadataTab GetSelectedTab()
+        {
+            return ConvertFromVm(SelectedTab);
+        }
+
+        public async Task InitializeAsync(DetailControlModel detailControlModel, INuGetPackageFileService nugetPackageFileService, ItemFilter currentFilter, PackageMetadataTab initialSelectedTab)
+        {
+            var nuGetFeatureFlagService = await ServiceLocator.GetComponentModelServiceAsync<INuGetFeatureFlagService>();
+            _readmeTabEnabled = await nuGetFeatureFlagService.IsFeatureEnabledAsync(NuGetFeatureFlagConstants.RenderReadmeInPMUI);
+
+            ReadmePreviewViewModel = new ReadmePreviewViewModel(nugetPackageFileService, currentFilter, _readmeTabEnabled);
+            DetailControlModel = detailControlModel;
+
+            Tabs.Add(ReadmePreviewViewModel);
+            Tabs.Add(DetailControlModel);
+
+            SelectedTab = Tabs.FirstOrDefault(t => t.IsVisible && ConvertFromVm(t) == initialSelectedTab) ?? Tabs.FirstOrDefault(t => t.IsVisible);
+
+            DetailControlModel.PropertyChanged += DetailControlModel_PropertyChanged;
+            ReadmePreviewViewModel.PropertyChanged += IsVisible_PropertyChanged;
```

Comment:
If we just hook up all tabs to the property changed event handler method that processes visibility, we won't have to change this code in the future if we add more tabs or change the logic so the other tabs become invisible, etc.

```suggestion
            foreach (var tab in Tabs)
            {
                tab.PropertyChanged += TabIsVisible_PropertyChanged;
            }
```

Reply:
If we do this, we need to unregister them as well, as not all these tab VMs are created by this class.